### PR TITLE
fix license headers and verify boilerplate

### LINKED ISSUE #1022: hack/verify: verify boilerplate
## Feature Request

**Describe the feature you'd like:**

Make sure all our files have the PingCAP author info.

### PR DESCRIPTION
**Pull request by cofyc**
### What problem does this PR solve? 

fixes #1022 

### Check List 

Tests 

 - Unit test
 - E2E test

### CODE REVIEW

File: hack/verify-boilerplate.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		-o -path './vendor/*' \
+		-o -path './pkg/client/*' \
+        -o -path './*/.terraform/*' \
+        -o -path './deploy/*' \
+        -o -path '*/*.sh' \
+        -o -path '*/Makefile' \
+        -o -path '*/Dockerfile' \
+        -o -path '*/*.register.go' \
+    \)
+))
```
Comment (aylei) [inline hack/verify-boilerplate.sh]: can we consider gitignore? it is okay for CI but broken in my local development env

Comment (cofyc) [reply hack/verify-boilerplate.sh]: They use different syntaxes. Which pattern you want to ignore, I can add it.

Comment (aylei) [reply hack/verify-boilerplate.sh]: `./.idea/*`

File: hack/boilerplate/boilerplate.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    return True
+
+
+def file_extension(filename):
+    return os.path.splitext(filename)[1].split(".")[-1].lower()
+
+
+skipped_dirs = ['Godeps', 'third_party', '_gopath', '_output', '.git', 'cluster/env.sh',
+                "vendor", "test/e2e/generated/bindata.go", "hack/boilerplate/test",
+                "staging/src/k8s.io/kubectl/pkg/generated/bindata.go"]
```
Comment (weekface) [inline hack/boilerplate/boilerplate.py]: do we have this directory?

Comment (cofyc) [reply hack/boilerplate/boilerplate.py]: these directories are hardcoded in this k8s script, we don't have these directories, it's ok to remove or keep them

File: hack/verify-boilerplate.sh (hunk 1 above)
Comment (cofyc) [reply hack/verify-boilerplate.sh]: fixed in f7306b8c67a5716b08eb7eff779d08ba671b2db6, all hidden files/directories are ignored

File: hack/boilerplate/boilerplate.py (hunk 2 above)
Comment (cofyc) [reply hack/boilerplate/boilerplate.py]: fixed in e095abbe072cfdd720c6c1ab560c352cd59e053c, hardcoded rules are ignored (except the script itself)

**Pull request merged**